server/datastore/mysql/hosts.go

```diff
@@ -3819,22 +3819,23 @@ ON DUPLICATE KEY UPDATE
 	return err
 }
 
-func (ds *Datastore) SaveLUKSData(ctx context.Context, hostID uint, encryptedBase64Passphrase string, encryptedBase64SlotKey string) error {
-	if encryptedBase64Passphrase == "" { // should have been caught at service level
-		return errors.New("blank encrypted passphrase")
+func (ds *Datastore) SaveLUKSData(ctx context.Context, hostID uint, encryptedBase64Passphrase string, encryptedBase64Salt string, keySlot uint) error {
```

Comment:
Nit: Should keySlot be `*uint` to map cleanly to the underlying NULLable type type.

Reply:
Nope. Only reason key slot is nullable in the DB is we use this table for BitLocker/FileVault, which (from our perspective at least) have neither key slot nor salt. Per the Slack convo earlier today, we're guaranteed to get key slot if we have valid LUKS data, and this method is only called when we have valid LUKS data, so we can tighten the parameter type.

Likewise we're guaranteed to have passphrase and salt when persisting LUKS, but I can't enforce that easily in the type system, hence the runtime check.